plenum/server/consensus/consensus_shared_data.py

```diff
@@ -74,6 +76,28 @@ def __init__(self, name: str, validators: List[str], inst_id: int, is_master: bo
         self.legacy_last_prepared_before_view_change = None
         self.prev_view_prepare_cert = None
 
+        # Dictionary of sent PRE-PREPARE that are stored by primary replica
+        # which it has broadcasted to all other non primary replicas
+        # Key of dictionary is a 2 element tuple with elements viewNo,
+        # pre-prepare seqNo and value is the received PRE-PREPARE
+        self.sent_preprepares = SortedDict(lambda k: (k[0], k[1]))
+        # type: Dict[Tuple[int, int], PrePrepare]
+
+        # Dictionary of all Prepare requests. Key of dictionary is a 2
+        # element tuple with elements viewNo, seqNo and value is a 2 element
+        # tuple containing request digest and set of sender node names(sender
+        # replica names in case of multiple protocol instances)
+        # (viewNo, seqNo) -> ((identifier, reqId), {senders})
+        self.prepares = Prepares()
+        # type: Dict[Tuple[int, int], Tuple[Tuple[str, int], Set[str]]]
+
+        self.commits = Commits()
+        # type: Dict[Tuple[int, int], Tuple[Tuple[str, int], Set[str]]]
+
+        # Tracks for which keys PRE-PREPAREs have been requested.
+        # Cleared in `gc`
+        self.requested_pre_prepares = {}
```

Comment:
Should we move `prePrepares` here as well?

Reply:
No, because OrderingService uses it.